python_modules/dagster-graphql/dagster_graphql/implementation/execution/launch_execution.py

```diff
@@ -53,3 +55,44 @@ def _launch_pipeline_execution(graphene_info, execution_params, is_reexecuted=Fa
     records = graphene_info.context.instance.get_run_records(RunsFilter(run_ids=[run.run_id]))
 
     return GrapheneLaunchRunSuccess(run=GrapheneRun(records[0]))
+
+
+@capture_error
+def launch_reexecution_from_parent_run(graphene_info, parent_run_id: str, policy):
+    """
+    Launch a re-execution by referencing the parent run id
+    """
+    from ...schema.inputs import GrapheneReexecutionPolicy
+    from ...schema.pipelines.pipeline import GrapheneRun
+    from ...schema.runs import GrapheneLaunchRunSuccess
+
+    check.inst_param(graphene_info, "graphene_info", ResolveInfo)
+    check.str_param(parent_run_id, "parent_run_id")
+
+    check.invariant(
+        policy == GrapheneReexecutionPolicy.FROM_FAILURE, "Only FROM_FAILURE is currently supported"
+    )
+
+    instance: DagsterInstance = graphene_info.context.instance
+    parent_run = instance.get_run_by_id(parent_run_id)
+    check.invariant(parent_run, "Could not find parent run with id: %s" % parent_run_id)
+
+    selector = PipelineSelector(
+        location_name=parent_run.external_pipeline_origin.external_repository_origin.repository_location_origin.location_name,
+        repository_name=parent_run.external_pipeline_origin.external_repository_origin.repository_name,
+        pipeline_name=parent_run.pipeline_name,
+        solid_selection=None,
+    )
+
+    repo_location = graphene_info.context.get_repository_location(selector.location_name)
+    external_pipeline = get_external_pipeline_or_raise(graphene_info, selector)
+
+    run = instance.create_reexecuted_run_from_failure(parent_run, repo_location, external_pipeline)
```

Comment:
We do miss the validation that `create_valid_pipeline_run` does, I think that's primarily checking run config against the external pipeline. Could add that to `create_reexecuted_run_from_failure`, but it seems not crucial, since it should only matter if the config schema has changed since the parent run?

Reply:
hmmm, that might be useful information though... does the user get a different error message before this change vs after?

Reply:
The same error message will still be surfaced: https://github.com/dagster-io/dagster/blob/b2a6e5022d8375878c1bf817f6d196398f8ee58f/python_modules/dagster/dagster/core/system_config/objects.py#L150-L156

But it isn't wrapped in the Graphene error classes: https://github.com/dagster-io/dagster/blob/581166282ff12acdb6da0f4b10a94a438bb85943/python_modules/dagster-graphql/dagster_graphql/implementation/external.py#L117-L124

I'm not sure what the best way would be if we did want to wrap those errors. Ideally the gql layer wouldn't need to fetch the external pipeline etc.

Reply:
Added a test case for invalid config

Reply:
There's a `@capture_error` decorator on the graphql implementation function, so I guess it will display the error message, but just as a PythonError and not this other error type.  Just based on `handleJobResult`, I think it's very similar treatment, but surfaces specific config errors to be resolved.  Ultimately, I think this is much less actionable in the reexecution case so I think this is probably okay?